### PR TITLE
Apikey review updates

### PR DESCRIPTION
**Pull request by davidmenendez**
Contributes to #954 

address the issues brought up in the review of ApiKeyModal with two major inclusions including adding an edit mode and refactoring certain functions to use `useEffect` instead. i also removed all the prop comments, which i will fix in another PR as well as finishing testing for editing.

### CODE REVIEW

**Comment by davidmenendez** [comment]
cc @dcwarwick @lee-chase

File: packages/cloud-cognitive/src/components/APIKeyModal/APIKeyModal.js [hunk 1]
```diff
@@ -59,66 +66,105 @@ export let APIKeyModal = forwardRef(
       onRequestSubmit,
       open,
       previousStepButtonText,
-      secondaryButtonText,
-      successBody,
-      successHeader,
-      stepHeaders,
+      secondaryButtonText: closeButtonText,
+      showPasswordLabel,
+      visibilityToggle,
```
Comment (lee-chase) [inline packages/cloud-cognitive/src/components/APIKeyModal/APIKeyModal.js:71]: Review requested this was renamed to `hasAPIKeyVisibilityToggle`

File: packages/cloud-cognitive/src/components/APIKeyModal/APIKeyModal.js [hunk 2]
```diff
@@ -236,174 +282,83 @@ export let APIKeyModal = forwardRef(
...
+            disabled={isPrimaryButtonDisabled}>
+            {primaryButtonText}
           </Button>
         </ModalFooter>
       </ComposedModal>
     );
   }
 );
 
+const customStepsRequiredProps = isRequiredIf(
```
Comment (lee-chase) [inline packages/cloud-cognitive/src/components/APIKeyModal/APIKeyModal.js:301]: Using PropTypes.string.isRequired.if directly in propTypes will add an * to the field in storybook documentation. Also worth adding a comment/note to the comment in propTypes e.g. NOTE: This prop is only required if customSteps are supplied.

File: packages/cloud-cognitive/src/components/APIKeyModal/APIKeyModal.js [hunk 3]
```diff
@@ -236,174 +282,83 @@ export let APIKeyModal = forwardRef(
...
+  PropTypes.string,
+  ({ customSteps }) => customSteps.length > 1
```
Comment (lee-chase) [inline packages/cloud-cognitive/src/components/APIKeyModal/APIKeyModal.js:303]:
```suggestion
  ({ customSteps }) => customSteps && customSteps.length > 1
```

**Review by lee-chase — CHANGES_REQUESTED** [review]

**Comment by davidmenendez** [comment]
@lee-chase @dcwarwick changes made

**Comment by davidmenendez** [comment]
@lee-chase @dcwarwick i was giving these changes some additional thought in regards to the use of `useEffect` and i have concluded that the original code was optimal as it was. i did some further research and using `useEffect` here to derive conditionals from props doesn't have any additional benefit and in reality causes additional unnecessary rerenders. any props that are going to effect the state of the modal are already going to cause a rerender, so we don't need to keep track of it in our own state. it's perfectly fine to just use standard variables here.

**Review by lee-chase — APPROVED** [review]

**Pull request merged**